codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/http/RequestBindingGenerator.kt

```diff
@@ -91,7 +91,7 @@ class RequestBindingGenerator(
      */
     fun renderUpdateHttpBuilder(implBlockWriter: RustWriter) {
         uriBase(implBlockWriter)
-        val addHeadersFn = httpBindingGenerator.generateAddHeadersFn(operationShape)
+        val addHeadersFn = httpBindingGenerator.generateAddHeadersFn(operationShape, serializeEmptyHeaders = true)
```

Comment:
Why does this only affect clients?

Reply:
The new tests for this only apply to clients: https://github.com/smithy-lang/smithy/pull/2403/files

There are similar tests for the server, but they still indicate that empty headers should not be serialized by the server: https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restJson1/http-headers.smithy#L455

I asked internally about this discrepancy and there didn't seem to be much of a reason for it, so it might be updated in the future to align the two.

Reply:
I made a PR to smithy to update the server tests to align: https://github.com/smithy-lang/smithy/pull/2433